src/commonTypes.ts

```diff
@@ -47,14 +53,17 @@ export type PageFile<FileType> = {
 
 export type PageParams = ParsedUrlQuery;
 
-export type PageObject = {
-  page: PageFile<NextPageFile>;
-  appFile: PageFile<NextAppFile>;
+export type RouteData = {
```

Comment:
Splitting this from `PageObject`  is not required for this PR, but it will enable easier implementation of the custom `_error` page